frontend/packages/console-dynamic-plugin-sdk/src/utils/fetch/console-fetch-utils.ts

```diff
@@ -46,3 +47,27 @@ export const getConsoleRequestHeaders = (): ConsoleRequestHeaders => {
 
   return headers;
 };
+
+export const parseData = async (response) => {
+  const text = await response.text();
+  const isPlainText = response.headers.get('Content-Type') === 'text/plain';
+  if (!text) {
+    return isPlainText ? '' : {};
+  }
+  return isPlainText || !response.ok ? text : JSON.parse(text);
+};
+
+/**
+ * This function takes a response object, converts the response to JSON, and then returns the data, headers, and status.
```

Comment:
The response is not always JSON. If it's JSON, it converts it to an object. If it's plaintext, it returns the plaintext.